packages/core/base/src/standard.ts

```diff
@@ -0,0 +1,33 @@
+import type {
+    SolanaSignAndSendTransactionFeature,
+    SolanaSignMessageFeature,
+    SolanaSignTransactionFeature,
+} from '@solana/wallet-standard-features';
+import type { Wallet as StandardWallet, WalletWithFeatures as StandardWalletWithFeatures } from '@wallet-standard/base';
+import type { ConnectFeature, DisconnectFeature, EventsFeature } from '@wallet-standard/features';
+import type { WalletAdapter, WalletAdapterProps } from './adapter.js';
+
+export type WalletAdapterCompatibleStandardWallet = StandardWalletWithFeatures<
+    ConnectFeature &
+        EventsFeature &
+        (SolanaSignAndSendTransactionFeature | SolanaSignTransactionFeature) &
+        (DisconnectFeature | SolanaSignMessageFeature | never)
+>;
```

Comment:
What does the `never` do in this type?

Reply:
This diff just moves the existing type into here from `@solana/wallet-standard-wallet-adapter-base`.

What this type is _trying_ to express (perhaps incorrectly) is that

- `EventsFeature` is required
- At least one of `SolanaSignAndSendTransactionFeature` or `SolanaSignTransactionFeature` is required (not mutually exclusive)
- `DisconnectFeature` and `SolanaSignMessageFeature` are optional (not mutually exclusive)

Reply:
Some testing suggests we actually want `(DisconnectFeature | SolanaSignMessageFeature | {})`:

https://www.typescriptlang.org/play?#code/LAKALgngDgpgBAJRgRwK4EsBOMAmBGOAXjgG84AibNLXPcgLjjE1XgF9RJY4B5KMdAHsAdgEMANgWJlyg-kLGSGTFu07R4fASIkAmIqQpzti3cuas4HEOu4BVYQoDq6MAAsAcjABuMTAaRqbHw4ADI4AAotBQkCAB9eeR1xfQThHz8ASlAAYxEAZzA4YQMSNjhRfLgHZ1dPDMwAblBQYQA6KgxgvBaQdAAzSIByYxjJIbh0EuFM0lA4BeK20eSekGsYcXz4EnnFgHp9uAABMHyAWhgAD1gcsEvMTEFMRgB3eFfRYSL3dCqwQRwfqidDiODnc5wABWqEKcAARjAcqJYfAAAYrRR4NGTfLCIY-Nx+eA4QQwPEEyYAWyg4ggcAxSVMOL+ewW7UxelA1lAA2GnJSEymxVmuxAiyWAt03N6XHgNRELncAFEaZAAigurQwpFoqs4Ak9aYDaQ2NkQHlhHD4NJypVqo5FXVVfwIM0bCAYB1NTR8L0+RERkzYkKSjBRWy4F6BWsNlsdpHDiczpcbkj7n4ni84O8c19CX8mIDgaDwZCYXDEcjUQyYyyKYTiXBSeT8UV0DS6bXgyl65Hoz3pet-YNA1LQ1GI+LFgOTFzhx6gA

Reply:
TypeScript Eslint [doesn't like this](https://typescript-eslint.io/rules/ban-types/#options) empty Object type (this why I avoided it originally, IIRC). It looks like `(DisconnectFeature | SolanaSignMessageFeature | object)` has the desired result though.

Reply:
Ah got you! Not worth changing in this PR, but I wonder if it could be defined as extending the required ones (only) so that we wouldn't need to add future optional features to the type. Probably fiddly/impossible to do in the generic, but probably worth considering if we have to extend it in future.

Reply:
Ah, we can see if we remove the casting that `never` is indeed a problem:

```ts
type Required1 = { required1: true }
type Optional1 = { optional1: true }
type Optional2 = { optional2: true }

type UnionWithNever = Required1 & (Optional1 | Optional2 | never)
const n: UnionWithNever = { required1: true };
```

<img width="784" alt="Screen Shot 2023-02-17 at 12 09 58 PM" src="https://user-images.githubusercontent.com/1173161/219743958-3b17bd61-88b3-4435-9b7c-acc681be78b9.png">

`never` is actually being elided, and we're left with a type of `Required1 & (Optional1 | Optional2)`, which means that our optional fields are becoming _at least one of required_ fields.

This is fixed using `object`:

```ts
type UnionWithObject = Required1 & (Optional1 | Optional2 | object)
const e: UnionWithObject = { required1: true };
```

Reply:
Fixed by changing `never` to `object`.

Reply:
> if it could be defined as extending the required ones (only) so that we wouldn't need to add future optional features to the type

The optional types have a purpose though. If we don't add them, we can't feature detect for them with type safety. We want the narrowing to express "this feature is optional, but if provided, it must have this type, and not another type".

https://www.typescriptlang.org/play?#code/LAKALgngDgpgBAJRgRwK4EsBOMAmBGOAXjgG85s0tc8AuOMTVeAX1EljgHkox0B7AHYBDADYFiZPj37CxdBkzitw0eN16DRAJiKk4UjbK3zGLUG1VwAqgJkB1dGAAWnAEYArGAGMwupJWx8OAAyOAAKdRlRAgAfLmlNER04vg9vMABKUAB6bLgAATAAZwBaGAAPWB8yzEw+TDoAd3hGoQFfZ3Qi+j44ADMhdBE4EpK4AAMDKLFxuC76JxhsABo4V1RfRwByboE+DsXydABzJw7VUC9BIt8YOht7Rxc0n10yCgxA2npTVanE74DERFFgAbnMIBgADoPlR8BD0H1wlt-rI8Fs5gI4DAMqRQHACdioajoqBlDBgfASPjCbkCsUypV0jU6g04M12W0DvMwL0BkMRmN3KgbmtvEIRfBJgk0bMugItgclvAcHwYEUFZsALZQEQQCYkpJyoo0gnQw1aMkIpFhFEy7QY9BYnF4kCEokWq0gUBAA